src/httpserver.cpp

```diff
@@ -419,6 +419,15 @@ bool InitHTTPServer()
     int workQueueDepth = std::max((long)gArgs.GetIntArg("-rpcworkqueue", DEFAULT_HTTP_WORKQUEUE), 1L);
     LogPrintfCategory(BCLog::HTTP, "creating work queue of depth %d\n", workQueueDepth);
 
+#if LIBEVENT_VERSION_NUMBER >= 0x02020001
+    if (event_get_version_number() >= 0x02020001) {
```

Comment:
Won't runtime linking fail to find the `evhttp_set_max_connections` symbol?

Reply:
Hm, how would that happen? I didn't have any issues when testing this code with 2.2.1-alpha and 2.1.12-stable.

Reply:
I'm thinking of the case where you build with 2.2.1, then try to run it with 2.1.12. (If that's not supported, the runtime check is unnecessary)